### ISSUE TITLE
CrossRef DOI lookup doesn't work

### ISSUE THREAD

**Issue by atomrab** [periodo/periodo-client#518]
When I provide a valid CrossRef DOI (checked against CrossRef itself, using the CrossRef URL provided for the item), instead of loading details, the local datasource "add authority" form allows the load button to be pressed, but then returns the message "failed to fetch source". See screenshot below:

<img width="1344" alt="Screen Shot 2023-09-08 at 12 46 33 PM" src="https://github.com/periodo/periodo-client/assets/4740995/ac45761b-35ec-4fc8-b9aa-544821ad0676">

Tested on Google Chrome Version 116.0.5845.179 on a MacBook Pro running Monterey 12.6.8.

**Comment by rybesh** [comment]
Looks like a CORS issue at CrossRef:

```
[Error] Origin https://client.perio.do is not allowed by Access-Control-Allow-Origin. Status code: 503
[Error] Failed to load resource: Origin https://client.perio.do is not allowed by Access-Control-Allow-Origin. Status code: 503 (15910, line 0)
[Error] Fetch API cannot load https://data.crossref.org/10.6092/issn.1971-8853/15910 due to access control checks.
```

Or maybe it’s just down?

```
$ curl -i https://data.crossref.org/10.6092/issn.1971-8853/15910
HTTP/1.1 503 Service Unavailable
content-length: 107
cache-control: no-cache
content-type: text/html
connection: close

<html><body><h1>503 Service Unavailable</h1>
No server is available to handle this request.
</body></html>
```

**Comment by rybesh** [comment]
Looks like we hard-coded some assumptions about the CrossRef URL space that no longer hold:

https://github.com/periodo/periodo-client/blob/2bcea582d86ed90b2aebef03cac42ba3a6eab0ba/modules/periodo-app/src/linked-data/utils/source_ld_match.js#L47

**Comment by rybesh** [comment]
CrossRef DOI (and WorldCat) lookups are fixed in v5.0.15, which is now deployed.
